xc/common/utils/prjxray_create_synth_tiles.py

```diff
@@ -70,33 +179,50 @@ def main():
 
     with DatabaseCache(args.connection_database, read_only=True) as conn:
         synth_tiles['info'] = j['info']
-        vbrk_in_use = set()
+        tile_in_use = set()
         for port in j['ports']:
-            if port['name'].startswith('dout['):
-                port_type = 'input'
-                is_clock = False
-            elif port['name'].startswith('din['):
-                is_clock = False
-                port_type = 'output'
-            elif port['name'].startswith('clk'):
-                port_type = 'output'
-                is_clock = True
+            if args.roi:
+                if port['name'].startswith('dout['):
+                    port_type = 'input'
+                    is_clock = False
+                elif port['name'].startswith('din['):
+                    is_clock = False
+                    port_type = 'output'
+                elif port['name'].startswith('clk'):
+                    port_type = 'output'
+                    is_clock = True
+                else:
+                    assert False, port
             else:
-                assert False, port
-
-            tile, wire = port['wire'].split('/')
+                if port['type'] == 'out':
+                    port_type = 'input'
+                    is_clock = False
+                elif port['type'] == 'in':
+                    is_clock = False
+                    port_type = 'output'
+                elif port['type'] == 'clk':
+                    port_type = 'output'
+                    is_clock = True
+                else:
+                    assert False, port
+
+            if args.partition_region:
+                tile, wire = find_wire_from_node(conn, g, roi, port['node'])
+            else:
+                tile, wire = port['wire'].split('/')
 
-            vbrk_in_use.add(tile)
+            tile_in_use.add(tile)
 
             # Make sure connecting wire is not in ROI!
             loc = g.loc_of_tilename(tile)
+            print(loc)
```

Comment:
Probably want to remove debug prints, or atleast put them behind a verbose flag?